### ISSUE TITLE
Enable editable code block fields

### ISSUE THREAD

**Issue by plumbis** [crossplane/docs#290]
Dynamically add `contentEditable="true"` to elements like the hover highlight function to allow editing elements before copying them.

https://www.tutorialrepublic.com/faq/how-to-make-a-div-element-editable-in-html.php

**Comment by plumbis** [comment]
I have this working in a private branch with a shortcode. I'll send a PR that includes updating the docs contrib guide after the docs merge.

It works by having a normal code block but using wrapping the element to edit with`$` 

For example:
````
{{<editCode >}}
```yaml {label="xr"}
cat <<EOF | kubectl apply -f -
apiVersion: s3.aws.upbound.io/v1beta1
kind: Bucket
metadata:
  name: $bucket$
spec:
  forProvider:
    region: us-east-2
  providerConfigRef:
    name: default
EOF
```
{{</editCode >}}
````

Here is a preview video:
https://user-images.githubusercontent.com/10537576/215880268-196c981d-858d-47dd-8187-14e3462c6bae.mov

**Comment by plumbis** [comment]
Merged in #311
